mmedit/datasets/pipelines/matlab_resize.py

```diff
@@ -0,0 +1,270 @@
+
+        def h(x):
+            return scale * kernel(scale * x)
+
+        kernel_width = 1.0 * kernel_width / scale
+    else:
+        h = kernel
+        kernel_width = kernel_width
+
+    # coordinates of output
+    x = np.arange(1, output_length + 1).astype(np.float32)
+
+    # coordinates of input
+    u = x / scale + 0.5 * (1 - 1 / scale)
+    left = np.floor(u - kernel_width / 2)  # leftmost pixel
+    p = int(ceil(kernel_width)) + 2  # maximum number of pixels
+
+    # indices of input pixels
+    ind = np.expand_dims(left, axis=1) + np.arange(p) - 1
+    indices = ind.astype(np.int32)
+
+    # weights of input pixels
+    weights = h(np.expand_dims(u, axis=1) - indices - 1)
+    weights = np.divide(weights,
+                        np.expand_dims(np.sum(weights, axis=1), axis=1))
+
+    # remove all-zero columns
+    aux = np.concatenate(
+        (np.arange(input_length), np.arange(input_length - 1, -1,
+                                            step=-1))).astype(np.int32)
+    indices = aux[np.mod(indices, aux.size)]
+    ind2store = np.nonzero(np.any(weights, axis=0))
+    weights = weights[:, ind2store]
+    indices = indices[:, ind2store]
+
+    return weights, indices
+
+
+def resize_along_dim(img_in, weights, indices, dim):
+    """Resize along a specific dimension.
+
+    Args:
+        img_in (ndarray): The input image.
+        weights (ndarray): The weights used for interpolation, computed from
+            [get_weights_indices].
+        indices (ndarray): The indices used for interpolation, computed from
+            [get_weights_indices].
+        dim (int): Which dimension to undergo interpolation.
+
+    Returns:
+        ndarray: Interpolated (along one dimension) image.
+    """
+
+    img_in = img_in.astype(np.float32)
+    w_shape = weights.shape
+    output_shape = list(img_in.shape)
+    output_shape[dim] = w_shape[0]
+    img_out = np.zeros(output_shape)
+
+    if dim == 0:
+        for i in range(w_shape[0]):
+            w = np.expand_dims(weights[i, :], 0)
+            ind = indices[i, :]
+            img_slice = img_in[ind, :]
+            img_out[i] = np.sum(
+                np.multiply(np.squeeze(img_slice, axis=0), w.T), axis=0)
+    elif dim == 1:
+        for i in range(w_shape[0]):
+            w = np.expand_dims(weights[i, :], 2)
+            ind = indices[i, :]
+            img_slice = img_in[:, ind]
+            img_out[:, i] = np.sum(
+                np.multiply(np.squeeze(img_slice, axis=1), w.T), axis=1)
+
+    if img_in.dtype == np.uint8:
+        img_out = np.clip(img_out, 0, 255)
+        return np.around(img_out).astype(np.uint8)
+    else:
+        return img_out
+
+
+@PIPELINES.register_module()
+class MATLABResize:
+    """Resize the input image using MATLAB downsampling.
+
+        Currently support bicubic interpolation only. Note that the output of
+        this function is slightly different from the official MATLAB function.
+
```

Comment:
Give a note on the added/modified keys like others https://github.com/open-mmlab/mmediting/blob/master/mmedit/datasets/pipelines/augmentation.py#L26-L28